src/rgw/rgw_data_sync.cc

```diff
@@ -2484,6 +2506,18 @@ int RGWBucketShardIncrementalSyncCR::operate()
           squash_entry = make_pair<>(e.timestamp, e.op);
         }
       }
+      if (syncstopped) {
+	drain_all();
+
+	yield {
+	  string oid = RGWBucketSyncStatusManager::status_oid(sync_env->source_zone, bs);
+	  RGWRados *store = sync_env->store;
+	  call(new RGWSimpleRadosDeleteCR(sync_env->async_rados, store, store->get_zone_params().log_pool, oid));
+	}
+
+	return set_cr_done();
+      }
+
```

Comment:
i like how you're reading past the `CLS_RGW_OP_SYNCSTOP` in case there's a `RESYNC` entry shortly after. but the definition of 'shortly after' here is kind of arbitrary - they'd have to be in the same batch of entries returned by `RGWListBucketIndexLogCR`. i think it'd be better to run incremental sync all the way to completion before doing this part. if sync is still disabled on this bucket, then we'll probably only see a couple entries after `SYNCSTOP`, if any. but if `RESYNC` comes anywhere later in the log, we really do want to keep reading and syncing entries